File: Ruby/casilla_calle.rb
require_relative "titulo_propiedad"

module Civitas
  class CasillaCalle < Casilla
    attr_reader :titulo
    
    def initialize(titulo)
      @titulo = titulo
      super(@titulo.nombre)
    end
    
    def recibe_jugador(actual, todos)
      if(jugador_correcto(actual, todos))
        super
        jugador = todos[actual]
        if(!@titulo.tienePropietario())
          jugador.puedeComprarCasilla
        else
          @titulo.tramitarAlquiler(jugador)
        end
        todo[actual] = jugador
      end
    end
    
    def to_s
      return super + "\nPropiedad " + @titulo.toString() + "\n"
    end
  end
end

File: Java/CasillaCalle.java
package civitas;

import java.util.ArrayList;

public class CasillaCalle extends Casilla {

    private float importe;
    private TituloPropiedad titulo;

    public CasillaCalle(TituloPropiedad tit) {
        super(tit.getNombre());
        titulo = tit;
        importe = tit.getPrecioCompra();
    }

    public TituloPropiedad getTituloPropiedad() {
        return titulo;
    }

    @Override
    public String toString() {
        return super.toString() + "\nPropiedad: " + titulo.toString();
    }

    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos) {
        if (super.jugadorCorrecto(actual, todos)) {
            super.informe(actual, todos);
            Jugador jugador = todos.get(actual);
            if (!titulo.tienePropietario()) {
                jugador.puedeComprarCasilla();
            } else {
                titulo.tramitarAlquiler(jugador);
            }
        }
    }

}
